src/Common/src/CoreLib/System/IO/PathInternal.cs

```diff
@@ -117,24 +117,54 @@ internal static bool AreRootsEqual(string first, string second, StringComparison
         internal static string RemoveRelativeSegments(string path, int rootLength)
         {
             Debug.Assert(rootLength > 0);
-            bool flippedSeparator = false;
+            
+            Span<char> initialBuffer = stackalloc char[260 /* PathInternal.MaxShortPath */];
+            ValueStringBuilder sb = new ValueStringBuilder(initialBuffer);
+
+            bool flippedSeparator = RemoveRelativeSegmentsHelper(path.AsSpan(), rootLength, ref sb);
+
+            // If we haven't changed the source path, return the original
+            if (!flippedSeparator && sb.Length == path.Length)
+            {
+                sb.Dispose();
+                return path;
+            }
+
+            return sb.Length < rootLength ? path.Substring(0, rootLength) : sb.ToString();
+        }
```

Comment:
Is this code used by any product code in corefx?  Seems like the change should be made in coreclr instead?

Reply:
The unit tests are here, which is one argument for making the change here. Whichever way you go, @Anipik, you should make sure you test with an updated CoreCLR in CoreFX.